src/HttpServerFactory.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive-swoole for the canonical source repository
+ * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive-swoole/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Zend\Expressive\Swoole;
+
+use Psr\Container\ContainerInterface;
+use Swoole\Http\Server as SwooleHttpServer;
+use SwooleRuntime;
+
+use function defined;
+use function in_array;
+
+use const SWOOLE_BASE;
+use const SWOOLE_PROCESS;
+use const SWOOLE_SOCK_TCP;
+use const SWOOLE_SOCK_TCP6;
+use const SWOOLE_SOCK_UDP;
+use const SWOOLE_SOCK_UDP6;
+use const SWOOLE_SSL;
+use const SWOOLE_UNIX_DGRAM;
+use const SWOOLE_UNIX_STREAM;
+
+class HttpServerFactory
+{
+    public const DEFAULT_HOST = '127.0.0.1';
+    public const DEFAULT_PORT = 8080;
+
+    /**
+     * Swoole server supported modes
+     */
+    public const MODES = [
+        SWOOLE_BASE,
+        SWOOLE_PROCESS
+    ];
+
+    /**
+     * Swoole server supported protocols
+     */
+    public const PROTOCOLS = [
```

Comment:
Given that these protocols are modified at runtime for the purpose of checking SSL support, would it be a good idea to make this a `private const`?

Reply:
If they are not consulted anywhere other than the factory, and/or there is no need for a user to know about the constants defined by the factory, `private` is fine.